lib/resources/shadow.rb

```diff
@@ -121,10 +120,14 @@ def to_s
       "/etc/shadow#{f}"
     end
 
+    def params
+      @params = @lines.nil? ? [] : @lines.map { |l| parse_shadow_line(l) }
+    end
+
     private
 
     def map_data(id)
-      @params.map { |x| x[id] }
+      params.map { |x| x[id] }
```

Comment:
Consider using `collect` here instead of `map` to make it clear that you're collecting a new set of values from the input data. (Same for the parsed shadow lines in the `params` method.)